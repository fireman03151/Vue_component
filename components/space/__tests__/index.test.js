import { mount } from '@vue/test-utils';
import Space from '..';
import mountTest from '../../../tests/shared/mountTest';

describe('Space', () => {
  mountTest(Space);

  it('should render width empty children', () => {
    const wrapper = mount({
      render() {
        return <Space />;
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render width customize size', () => {
    const wrapper = mount({
      render() {
        return (
          <Space size={10}>
            <span>1</span>
            <span>2</span>
          </Space>
        );
      },
    });
    expect(wrapper.find('.ant-space').element.style.columnGap).toBe('10px');
    expect(wrapper.find('.ant-space').element.style.rowGap).toBe('10px');
  });

  it('should render vertical space width customize size', () => {
    const wrapper = mount({
      render() {
        return (
          <Space size={10} direction="vertical">
            <span>1</span>
            <span>2</span>
          </Space>
        );
      },
    });
    expect(wrapper.find('.ant-space').element.style.columnGap).toBe('10px');
    expect(wrapper.find('.ant-space').element.style.rowGap).toBe('10px');
  });

  it('should render correct with children', () => {
    const wrapper = mount({
      render() {
        return (
          <Space>
            text1<span>text1</span>
            text3
          </Space>
        );
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render with invalidElement', () => {
    const wrapper = mount({
      render() {
        return (
          <Space>
            text1<span>text1</span>
            text1
          </Space>
        );
      },
    });

    expect(wrapper.findAll('.ant-space-item').length).toBe(3);
  });
});
