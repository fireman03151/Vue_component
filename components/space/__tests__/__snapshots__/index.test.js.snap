// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space should render correct with children 1`] = `
<div class="ant-space ant-space-horizontal ant-space-align-center" style="column-gap: 8px; row-gap: 8px;">
  <div class="ant-space-item">text1</div>
  <!---->
  <div class="ant-space-item"><span>text1</span></div>
  <!---->
  <div class="ant-space-item">text3</div>
  <!---->
</div>
`;

exports[`Space should render width empty children 1`] = `<!---->`;
